framework/lib/components/tags-input/tags-input.tsx

```diff
@@ -0,0 +1,170 @@
+import React, {
+  FunctionComponent,
+  KeyboardEvent,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import {
+  ActionMeta,
+  CreatableSelect,
+  GroupBase,
+  InputActionMeta,
+  MultiValue,
+  OptionsOrGroups,
+} from 'chakra-react-select'
+import { equals, identity } from 'ramda'
+import { Box } from '../box'
+import { useToken } from '../../hooks'
+import { TagsInputProps } from './types'
+import { Option } from '../select'
+import { tagsInputStyles } from './tags-input-styles'
+
+/**
+ * Tags Input, based of react creatable select, is meant to select
+ * multiple tags. Follows styling of textarea and is resizable.
+ *
+ * @see {@link https://northlight.dev/reference/tags-input}
+ * @example (Example)
+ * ## Simple
+ * (?
+ * <TagsInput />
+ * ?)
+ *
+ * @example (Example)
```

Comment:
When options are specified, disallow values that are outside of scope.
Optionally, provide a separate prop that would control this behaviour.

Reply:
maybe, but it still feels weird to have suggestions and allow any value to be saved at the same time.
discard this comment for now